src/reducers/tests/projects.reducer.test.js

```diff
@@ -0,0 +1,204 @@
+import {
+  IMPORT_EXISTING_PROJECT_FINISH,
+  ADD_DEPENDENCY_FINISH,
+  REFRESH_PROJECTS,
+  SELECT_PROJECT,
+} from '../../actions';
+
+import reducer, {
+  initialState as projectsInitialState,
+  getById,
+  getSelectedProjectId,
+  getInternalProjectById,
+  getProjectsArray,
+} from '../projects.reducer';
+
+describe('Projects Reducer', () => {
+  describe(IMPORT_EXISTING_PROJECT_FINISH, () => {
+    it('imports existing project', () => {
+      const action = {
+        type: IMPORT_EXISTING_PROJECT_FINISH,
+        project: {
+          name: 'testing',
+          guppy: { id: 'best-id' },
+          scripts: {
+            start: 'react-scripts start',
+          },
+        },
+      };
+      const actualState = reducer(projectsInitialState, action);
+
+      const { name, guppy, scripts } = action.project;
+      expect(actualState).toEqual({
+        byId: {
+          [guppy.id]: {
+            name,
+            guppy,
+            scripts,
+          },
+        },
+        selectedId: guppy.id,
+      });
+    });
+  });
```

Comment:
Can you add tests for ADD_PROJECT as well? It shares exactly the same effect as IMPORT_EXISTING_PROJECT_FINISH, so you could do something like:

```
const newProjectActions = [ADD_PROJECT, IMPORT_EXISTING_PROJECT_FINISH];

newProjectActions.forEach(ACTION => {
  describe(ACTION, () => {
    ... repeat all the logic you have here
  });
}
```